### ISSUE TITLE
i2t_sft code

### ISSUE THREAD

**Issue by ioky31** [baaivision/Emu3#65]
Can you provide fine-tuning code for image understanding

**Comment by ryanzhangfan** [comment]
You can easily adapt current train and dataset codes for image understanding tasks. Just modify the [datasets.py](https://github.com/baaivision/Emu3/blob/main/emu3/train/datasets.py.py) to accommodate image understanding data format and especially change labels to supervise text tokens only.

**Comment by ioky31** [comment]
May I ask if this change is correct?
 
 vocab = tokenizer.get_vocab()
  self.min_id = min(vocab.values())
  self.max_id = max(vocab.values())
if self.args.apply_loss_on_only_text: 
            labels = torch.where(torch.logical_and(labels >= self.min_id, labels <= self.max_id), labels, self.args.ignore_index)

**Comment by ioky31** [comment]
> You can easily adapt current train and dataset codes for image understanding tasks. Just modify the [datasets.py](https://github.com/baaivision/Emu3/blob/main/emu3/train/datasets.py.py) to accommodate image understanding data format and especially change labels to supervise text tokens only.

May I ask if this change is correct?

vocab = tokenizer.get_vocab()
self.min_id = min(vocab.values())
self.max_id = max(vocab.values())
if self.args.apply_loss_on_only_text: 
    labels = torch.where(torch.logical_and(labels >= self.min_id, labels <= self.max_id), labels, self.args.ignore_index)

**Comment by zhangzef** [comment]
@ryanzhangfan 哈喽，我注意到模型本身的dtype为float32，然后sft的代码加载模型时使用的是bfloat16，我经过了四个epoch的训练，模型在前三个epoch无法输出正常的文本，解码后所有的token均是"0"，第四个epoch后模型可以正常进行输出，但是似乎效果不是特别好，麻烦问一下这是不是由于数值类型转换导致的数值溢出呀。
